### PR TITLE
Add `onCurrentIndexChange` prop to `LineChart`

### PR DESCRIPTION
**Pull request by nandorojo**

### CODE REVIEW

**Comment by nandorojo** [comment]
Should the animated reaction fire when x changes, or when `currentIndex` changes 🧐

**Comment by nandorojo** [comment]
I tested both; it should def be the `currentIndex` changes. So I fixed that.

**Comment by nandorojo** [comment]
I did remove `invokeHaptic` from the documented recommendations. It made my Expo app crash when I invoked haptics a few hundred times back and forth. Maybe that's an edge case, but figured I wouldn't make it the recommendation in the docs themeslves.

File: src/charts/line/Context.tsx [hunk 1]
```diff
@@ -21,12 +25,14 @@ type LineChartProviderProps = {
   children: React.ReactNode;
   data: TLineChartData;
   yRange?: YRangeProp;
+  onCurrentXChange?: (x: number) => void;
```
Comment (jxom) [inline src/charts/line/Context.tsx:28]: Maybe we should name this `onCurrentIndexChange`?

Comment (nandorojo) [reply src/charts/line/Context.tsx]: Done

**Pull request merged**